guidelines/faster-review.md

```diff
@@ -0,0 +1,115 @@
+# Faster Code Review Guideline
+
+Kyma is quite a large project with many areas, each has a group of core developers
+who are working hard on developing that area of Kyma. While that means they are
+working hard on new features and enhancements, they also work on doing code
+reviews for PRs coming from different contributors from the community and it is
+crucial that the code review process of new PRs becomes quick and efficient for
+both parties. This guide defines a set of principles and practices that can help
+achieving such a goal.
+
+## 0. Familiarize yourself with project conventions
+
+- [CONTRIBUTING.md](../CONTRIBUTING.md)
+- [Naming conventions](naming.md)
+- [Content strategy](content-guidelines/content-strategy.md)
+- [Content guidelines](content-guidelines)
+- [Coding standards](coding-standards)
+
+## 1. Before coding communication
+
+We recommend these steps before you jump into coding that nice feature you want
+
+1. Make sure you have a GitHub issue created for the work you want to do that
+   - Follows one of the issues templates and make sure to describe well the
+    the changes you intend to introduce.
+   - If possible identify in the issue the relevant area label and components
+    you might touch and change.
+2. Communicate that issue upfront on the slack channel of the relevant area and
+start a discussion there to have it pre-reviewed.
+3. Ask the area group to assign you an issue buddy from the CODEOWNERS.
+    >Since reviewing a PR always needs context, it is best if the whole pre-review,
+assistance, and code review stays with a single area member who is also a
+CODEOWNER. We call that person an `issue buddy`.
+
```

Comment:
To me it is not quite clear to whom is the faster-review guideline here intended: to external contributors or to all contributors (including CODEOWNERs ). How to ask Area Group to assign a buddy? 

The steps above suggest one who creates an issue is also delivering the code for that issue and that it happens very soon after issue has been created.
 
Pre-review should happen just in time i.e. once the issue has been triaged and agreed to work upon. 
If the contributor of the issue wants the issue to be taken into triage asap, then of course, sufficient context needs to be provided. 
Note that code review starts once there is a code for the issue and not before. Perhaps the guideline needs to be splitted? Smth like. 
* Create an issue that matters  
* Get your PR reviewed properly

Step 2.  could be automated - and as soon the issue has been assigned an area.

Reply:
it is written to do it on slack, very good approach imho,
I don't think it is imposed that code comes very soon after issues creation

it is just a recommendation to the contributor that whenever he creates an issue for a given area and wants to contribute, he should contact area members and discuss, a kind of savoir-vivre